src/creditcard.js

```diff
@@ -37,6 +25,40 @@ export const isValid = number => {
   return sum > 0 && sum % 10 === 0;
 };
 
+function isValidMonth(month) {
+  return !isNaN(month) && month >= 1 && month <= 12;
+}
+
+function isValidYear(year) {
+  return !isNaN(year) && isValidFullYear(formatFullYear(year));
+}
+
+function formatFullYear(year) {
+  if (year.length === 4)
+    return year;
+
+  if (year.length === 2) {
+    return parseInt(year) + 2000;
```

Comment:
queria adotar aqui uma solução melhor que essa :thinking:

Reply:
Então, rola fazer algo como citaram aqui https://stackoverflow.com/questions/51254131/how-to-get-the-nearest-decade-century-millenium-with-javascript

Reply:
```js
const dateRange = (date, rule, sum = 0) => Math.floor(date.getFullYear() / rule) * rule + sum;

console.log(dateRange(new Date() , 10 /* a cada 10 anos */ )); //2020
console.log(dateRange(new Date() , 100 /* a cada 100 anos */ )); //2000
console.log(dateRange(new Date() , 100 /* a cada 1000 anos */ )); //2000
```

Reply:
massa! foi aplicar